commands/run.sh

```diff
@@ -348,9 +348,9 @@ fi
 
 if [[ "${BUILDKITE_PLUGIN_DOCKER_ALWAYS_PULL:-false}" =~ ^(true|on|1)$ ]] ; then
   echo "--- :docker: Pulling ${image}"
-  if ! retry "${BUILDKITE_PLUGIN_DOCKER_PULL_RETRIES:-3}" \
-       docker pull "${image}" ; then
-    retry_exit_status="$?"
+  retry "${BUILDKITE_PLUGIN_DOCKER_PULL_RETRIES:-3}" docker pull "${image}"
+  retry_exit_status="$?"
+  if [ $retry_exit_status -ne 0 ] ; then
```

Comment:
A `set -e`-friendly way of doing this would be:

```suggestion
  retry "${BUILDKITE_PLUGIN_DOCKER_PULL_RETRIES:-3}" docker pull "${image}" || retry_exit_status="$?"
  if [ ${retry_exit_status:-0} -ne 0 ] ; then
```